easybuild/easyblocks/m/mesa.py

```diff
@@ -40,42 +42,78 @@ class EB_Mesa(MesonNinja):
-            if get_cpu_architecture() == X86_64:
+        if 'swr' in gallium_drivers:
+            # Check user-defined SWR arches
+            self.swr_arches = self.get_configopt_value('swr-arches')
+
+            if not self.swr_arches:
+                # Set cpu features of SWR for current micro-architecture
                 feat_to_swrarch = {
                     'avx': 'avx',
                     'avx1.0': 'avx',  # on macOS, AVX is indicated with 'avx1.0' rather than 'avx'
                     'avx2': 'avx2',
                     'avx512f': 'skx',  # AVX-512 Foundation - introduced in Skylake
                     'avx512er': 'knl',  # AVX-512 Exponential and Reciprocal Instructions implemented in Knights Landing
                 }
-                # determine list of values to pass to swr-arches configuration option
+                # Determine list of values to pass to swr-arches configuration option
                 cpu_features = get_cpu_features()
                 self.swr_arches = sorted([swrarch for feat, swrarch in feat_to_swrarch.items() if feat in cpu_features])
+                # Add configopt for additional SWR arches
+                self.gallium_configopts.append('-Dswr-arches=' + ','.join(self.swr_arches))
+
+            self.log.debug('SWR Gallium driver will support: %s' % ', '.join(self.swr_arches))
+
+    def get_configopt_value(self, configopt_name):
+        """
+        Return list of values for the given configuration option in configopts
+        """
+        configopt_value = [opt.split('=')[1] for opt in self.cfg['configopts'].split() if configopt_name in opt]
+
+        if configopt_value:
+            configopt_value = configopt_value[-1].replace("'", '').replace('"', '')
+            configopt_value = configopt_value.split(',')
+
+            if len(configopt_value) > 1:
+                configopt_pair = "%s = %s" % (configopt_name, ', '.join(configopt_value))
+                self.log.warning(
+                    "Found multiple instances of %s in configopts, using last one: %s", configopt_name, configopt_pair
+                )
+
+        return configopt_value
 
     def configure_step(self):
         """
         Customise the configure options based on the processor architecture of the host
-        (x86_64 or not, CPU features, ...)
+        (Gallium drivers installed, SWR CPU features, ...)
         """
 
-        arch = get_cpu_architecture()
-        if 'gallium-drivers' not in self.cfg['configopts']:
-            # Install appropriate Gallium drivers for current architecture
-            if arch == X86_64:
-                self.cfg.update('configopts', "-Dgallium-drivers='swrast,swr'")
-            elif arch == POWER:
-                self.cfg.update('configopts', "-Dgallium-drivers='swrast'")
-
-        if self.swr_arches:
-            self.cfg.update('configopts', '-Dswr-arches=' + ','.join(self.swr_arches))
+        if self.gallium_configopts:
```

Comment:
This can be removed with suggestion in https://github.com/easybuilders/easybuild-easyblocks/pull/2006/files#diff-1cf1bc7f7b0deed686907a2e5a9e0a96R64

Reply:
See my reply above